From 356d9a3592996d0fdff2c20ecceb0683cd5fc33f Mon Sep 17 00:00:00 2001
From: Suneel Garapati <sgarapati@marvell.com>
Date: Thu, 21 Nov 2019 15:29:46 -0800
Subject: [PATCH 1011/1239] drivers: spi: octeontx: use read after write for
 MPI_CFG

Use read to make previous write observed for mode change
in config register.

Change-Id: I2537e606e16c19dab2aa0298e747080a1b101b36
Signed-off-by: Suneel Garapati <sgarapati@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/19110
Reviewed-by: Chandrakala Chavva <cchavva@marvell.com>
Tested-by: Chandrakala Chavva <cchavva@marvell.com>
(cherry picked from commit 32a941830ddedd8ee5912209ea86246fff06a398)
Reviewed-on: https://sj1git1.cavium.com/19365
Tested-by: Suneel Garapati <sgarapati@caviumnetworks.com>
Reviewed-by: Suneel Garapati <sgarapati@caviumnetworks.com>
---
 drivers/spi/octeontx_spi.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/spi/octeontx_spi.c b/drivers/spi/octeontx_spi.c
index 15df8d7c13..8214235f3f 100644
--- a/drivers/spi/octeontx_spi.c
+++ b/drivers/spi/octeontx_spi.c
@@ -310,6 +310,7 @@ static int octeontx_spi_claim_bus(struct udevice *dev)
 	mpi_cfg.s.tritx = 0;
 	mpi_cfg.s.enable = 1;
 	writeq(mpi_cfg.u, baseaddr + MPI_CFG);
+	mpi_cfg.u = readq(baseaddr + MPI_CFG);
 	udelay(5);	/** Wait for bus to settle */
 
 	return 0;
@@ -340,6 +341,7 @@ static int octeontx_spi_release_bus(struct udevice *dev)
 	mpi_cfg.u = readq(baseaddr + MPI_CFG);
 	mpi_cfg.s.enable = 0;
 	writeq(mpi_cfg.u, baseaddr + MPI_CFG);
+	mpi_cfg.u = readq(baseaddr + MPI_CFG);
 	udelay(1);
 
 	return 0;
@@ -371,6 +373,7 @@ static int octeontx_spi_xfer(struct udevice *dev, unsigned int bitlen,
 
 	if (mpi_cfg.u != readq(baseaddr + MPI_CFG)) {
 		writeq(mpi_cfg.u, baseaddr + MPI_CFG);
+		mpi_cfg.u = readq(baseaddr + MPI_CFG);
 		udelay(10);
 	}
 
@@ -480,6 +483,7 @@ static int octeontx_spi_xfer(struct udevice *dev, unsigned int bitlen,
 
 	if (mpi_cfg.u != readq(baseaddr + MPI_CFG)) {
 		writeq(mpi_cfg.u, baseaddr + MPI_CFG);
+		mpi_cfg.u = readq(baseaddr + MPI_CFG);
 		udelay(10);
 	}
 
-- 
2.29.0

